src/gsy_e/gsy_e_core/area_serializer.py

```diff
@@ -128,7 +128,7 @@ def optional(attr):
         if key_in_dict_and_not_none(description, 'children'):
             children = [area_from_dict(child, config) for child in description['children']]
         else:
-            children = None
+            children = []
```

Comment:
Actually it is allowed to be an empty list, in [this](https://github.com/gridsingularity/gsy-e/blob/29ce17dc01792ef41ed8b70322fda62d2364ed01/src/gsy_e/models/area/__init__.py#L111) we pass the children to the `AreaChildrenList` and this will basically apply such operation: `list([])`which will return [].

Reply:
Indeed, the empty list will be set to the area.children. However, in the place you posted, we are checking `if children is not None` which will never be true with this change here. 
So, IMO this needs to be changed, and also every occurrence where we check for the children of an area.
`if not area.children:` also works with `children=[]` bit if there is a place were we check like this: `if area.children is None` then it does not. 
If all agree, that we do this change from children=None to children=[] for market areas that do not have children, then please adapt and check the above.

Reply:
I'd agree with Hannes, since this apparently would generate some issues in other parts of the code.

In general, I think `if not foo.bar` should be preferred over `if foo.bar is not None`. The exception is e.g. when we have a very specific reason to distinguish between false values (like `False`, `0`, `""`, `[]`) and `None`.
In our case, however, I think that these 2 are equivalent: 
- `area.children = None`
- `area.children = []`

If this is true, checking for `if not area.children` seems to be the best way to proceed.